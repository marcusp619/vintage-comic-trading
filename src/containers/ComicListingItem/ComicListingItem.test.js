import React from "react";

import { shallow, mount } from "enzyme";

import { ComicListItem, mockStateToProps, mockDispatchToProps } from "./index";

describe("ComicListItem", () => {
  let mockArray;
  beforeAll(() => {
    mockArray = [
      {
        id: 1009718,
        name: "Wolverine",
        description:
          "Born with super-human senses and the power to heal from almost any wound, Wolverine was captured by a secret Canadian organization and given an unbreakable skeleton and claws. Treated like an animal, it took years for him to control himself. Now, he's a premiere member of both the X-Men and the Avengers."
      },
      {
        id: 1009718,
        name: "Wolverine",
        description:
          "Born with super-human senses and the power to heal from almost any wound, Wolverine was captured by a secret Canadian organization and given an unbreakable skeleton and claws. Treated like an animal, it took years for him to control himself. Now, he's a premiere member of both the X-Men and the Avengers."
      }
    ];
  });

  it("ComicListingItem should match snapshot", () => {
    const wrapper = shallow(<ComicListItem comic={mockArray} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should toggle displayComicData state", () => {
    const expectedState = {
      displayComicData: true
    };
    const wrapper = shallow(<ComicListItem comic={mockArray} />);
    wrapper.instance().toggleHoverState();

    expect(wrapper.state()).toEqual(expectedState);
  });
});
