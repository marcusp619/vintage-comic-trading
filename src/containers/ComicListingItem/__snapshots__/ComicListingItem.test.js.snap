// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComicListItem ComicListingItem should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ComicListItem
    comic={
      Array [
        Object {
          "description": "Born with super-human senses and the power to heal from almost any wound, Wolverine was captured by a secret Canadian organization and given an unbreakable skeleton and claws. Treated like an animal, it took years for him to control himself. Now, he's a premiere member of both the X-Men and the Avengers.",
          "id": 1009718,
          "name": "Wolverine",
        },
        Object {
          "description": "Born with super-human senses and the power to heal from almost any wound, Wolverine was captured by a secret Canadian organization and given an unbreakable skeleton and claws. Treated like an animal, it took years for him to control himself. Now, he's a premiere member of both the X-Men and the Avengers.",
          "id": 1009718,
          "name": "Wolverine",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          alt="comicbook cover"
          src="undefined"
        />,
        false,
      ],
      "onClick": [Function],
      "onMouseEnter": [Function],
      "onMouseLeave": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "comicbook cover",
          "src": "undefined",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bwzfXH",
            "isStatic": true,
            "rules": Array [
              "
  width: 100%;
  height: 100%;
",
            ],
          },
          "displayName": "styled.img",
          "render": [Function],
          "styledComponentId": "sc-bwzfXH",
          "target": "img",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      false,
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bdVaJa",
        "isStatic": true,
        "rules": Array [
          "
  box-shadow: 9px 14px 25px -5px rgba(0, 0, 0, 0.75);
  background: grey;
  height: 300px;
  position: relative;
  transition: all 0.2s ease-in-out;
  width: 250px;

  &:hover {
    transform: scale(1.1);
    z-index: 99;
  }
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bdVaJa",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            alt="comicbook cover"
            src="undefined"
          />,
          false,
        ],
        "onClick": [Function],
        "onMouseEnter": [Function],
        "onMouseLeave": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "comicbook cover",
            "src": "undefined",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bwzfXH",
              "isStatic": true,
              "rules": Array [
                "
  width: 100%;
  height: 100%;
",
              ],
            },
            "displayName": "styled.img",
            "render": [Function],
            "styledComponentId": "sc-bwzfXH",
            "target": "img",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        false,
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": true,
          "rules": Array [
            "
  box-shadow: 9px 14px 25px -5px rgba(0, 0, 0, 0.75);
  background: grey;
  height: 300px;
  position: relative;
  transition: all 0.2s ease-in-out;
  width: 250px;

  &:hover {
    transform: scale(1.1);
    z-index: 99;
  }
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
